### PR TITLE
Fix broken UI tests

### PR DESCRIPTION
**Pull request by marcussc**
## Description
Since we now have one scanner to rule all QRCodes instead of many, our UI-Tests starting to fail all over the place since each of the old QRCodesScanners had different "hacks" to get the Simulator to scan a QRCode.
When the QRCodeScanner is opened now in the Simulator you don't get an error message but an Alert where you can choose which QRCode you want to fake/scan. 
The failing tests are now using this alert to choose the right QRCode they need to be scanned. 
With 2.12 we would have the chance to remove this "hack" and use the image picker and equip the simulator with the needed QRCodes. This would improve our tested codepaths a lot (@KaiTeuber)

I also removed two performance tests that stole 2 minutes from our CI Runtime without providing any meaningful insights since the results never got tracked.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9627

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift [hunk 1]
```diff
@@ -348,4 +357,36 @@ class QRScannerViewController: UIViewController {
 		previewLayer.mask?.addSublayer(throughHoleLayer)
 		previewLayer.mask?.addSublayer(backdropLayer)
 	}
+
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:360]: And for warning on-behalf we do not need an action, correct?

File: src/xcode/ENA/ENA/Source/Client/Security/__tests__/CryptoFallbackTests.swift [hunk 2]
```diff
@@ -23,26 +23,6 @@ class CryptoFallbackTests: iOS13TestCase {
 		XCTAssertEqual(customDigest, referenceDigest)
     }
 
-    func testPerformanceCryptoKitSHA256() throws {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Client/Security/__tests__/CryptoFallbackTests.swift:26]: 🚮❤️

**Review by pascal-brause — COMMENTED** [review]
For all occurrences of `#if targetEnvironment(simulator)`: Was it intentionally or why not using `#if DEBUG if isUITesting {`?

**Comment by marcussc** [comment]
> For all occurrences of `#if targetEnvironment(simulator)`: Was it intentionally or why not using `#if DEBUG if isUITesting {`?

@pascal-brause I think `#if targetEnvironment(simulator)` is nicer since this enables us to test all flows that involve a scanned QRCode in the Simulator and not on the device. I also think this makes debugging easier for the Community. But as a mentioned this should get removed once move to 2.12 and we can just use QRCodes saved to the CameraRoll/Files.

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift (hunk 1 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:360]: You need an event for warning on-behalf :)

Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:360]: What I meant was that we do not need an UIActionSheet action here for scanning an event for the warning on-behalf?

Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:360]: There is one in line 382 🕵️‍♂️

Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift:360]: 🤦‍♂️

**Review by pascal-brause — APPROVED** [review]

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**